test/VaultOUSDLvUSD.js

```diff
@@ -0,0 +1,172 @@
+const { expect } = require("chai");
+const exp = require("constants");
+const { ethers } = require("hardhat");
+
+describe("VaultOUSDLvUSD test suit", function () {
+    let vaultToken;
+    let OUSDMockToken
+    let lvUSDToken;
+    let owner;
+    let addr1;
+    let addr2;
+    let addrs;
+
+    const defaultUserOUSDBalance = 1000000;
+    const defaultUserLvUSDBalance = 10000; /// Using different value for lvUSD and OUSD to distinguish 
+
+    beforeEach(async function () {
+        let contract = await ethers.getContractFactory("VaultOUSDLvUSD");
+        let mockOUSDContract = await ethers.getContractFactory("OUSDMockToken");
+        let lvUSDContract = await ethers.getContractFactory("LvUSDToken");
+        [owner, addr1, addr2, ...addrs] = await ethers.getSigners();
+        OUSDMockToken = await mockOUSDContract.deploy();
+        lvUSDToken = await lvUSDContract.deploy()
+        vaultToken = await contract.deploy(OUSDMockToken.address, lvUSDToken.address);
+
+        // Mint initial amount on OUSD token, will be used by all tests/
+        // "owner" is calling this mint function. addr1 is our client's address
+        await OUSDMockToken.mint(addr1.address, defaultUserOUSDBalance)
+        await lvUSDToken.mint(addr1.address, defaultUserLvUSDBalance)
+    });
+
+    async function depositOUSDFundsInVault(amount) {
+        balance = await vaultToken.getVaultOUSDBalance();
+        await depositOUSDFundsInVaultFromAccount(addr1, amount)
+        expect(await vaultToken.getVaultOUSDBalance()).to.equal(balance + amount)
+    };
+
+    async function depositOUSDFundsInVaultFromAccount(fromAccount, amount) {
+        // first set it to 0 to mitigate potential abuse https://ethereum.org/en/developers/tutorials/erc20-annotated-code/
+        await OUSDMockToken.connect(fromAccount).approve(vaultToken.address, 0);
+        await OUSDMockToken.connect(fromAccount).approve(vaultToken.address, amount);
+        // console.log("test:Vault:depositFunds:addr1 allowance %s", await)
+        await vaultToken.depositOUSD(fromAccount.address, amount);
+    };
+
+    async function checkVaultOUSDBalance(expected) {
+        let vaultBalance = await vaultToken.getVaultOUSDBalance();
+        expect(vaultBalance).to.equal(expected);
+        expect(await OUSDMockToken.balanceOf(vaultToken.address)).to.equal(expected)
+    };
+
+    async function depositLvUSDIntoVault(amount) {
+        await lvUSDToken.connect(addr1).approve(vaultToken.address, 0);
+        await lvUSDToken.connect(addr1).approve(vaultToken.address, amount);
+        await vaultToken.depositLvUSD(addr1.address, amount);
+    }
+
+    async function checkVaultLvUSDBalance(expected) {
+        expect(await lvUSDToken.balanceOf(vaultToken.address)).to.equal(expected)
+        expect(await vaultToken.getVaultLvUSDBalance()).to.equal(expected);
+
+    }
+
```

Comment:
So the above is the "helper function section" - below is the real deal. I have comment-philia I admit :-) but maybe add comment at the beginning of each section?